### PR TITLE
Merge Compose 1.5.1 release

### PR DESCRIPTION
**Pull request by Walingar**

### CODE REVIEW

File: compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/CoreTextField.kt [hunk 1]
```diff
@@ -301,13 +304,17 @@ internal fun CoreTextField(
...
 
         if (textInputService != null) {
-            notifyTextInputServiceOnFocusChange(
-                textInputService,
-                state,
-                value,
-                imeOptions,
-                offsetMapping
-            )
+            if (state.hasFocus && enabled && !readOnly) {
```
Comment (Walingar) [inline compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/CoreTextField.kt:307]: It seems this is the only thing that we should take a look

**Comment by eymar** [comment]
Were there any merge conflicts?

File: compose/foundation/foundation-layout/src/androidAndroidTest/kotlin/androidx/compose/foundation/layout/RowColumnTest.kt [hunk 2]
```diff
@@ -1692,6 +1692,34 @@ class RowColumnTest : LayoutTest() {
         assertTrue(latch.await(1, TimeUnit.SECONDS))
```
Comment (Walingar) [inline compose/foundation/foundation-layout/src/androidAndroidTest/kotlin/androidx/compose/foundation/layout/RowColumnTest.kt:1692]: @eymar there were simple merge conflicts: 
  1. build.gradle where version number of a library was changed
  2. import of kotlin.jvm.JvmInline in InlineDensityKt in commonMain (this should be upstreamed)
  3. androidTests

image

File: compose/compiler/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/VersionChecker.kt [hunk 3]
```diff
@@ -121,6 +121,7 @@ class VersionChecker(val context: IrPluginContext) {
             10400 to "1.5.0-beta03",
             10401 to "1.5.0-rc01",
             10402 to "1.5.0",
+            10403 to "1.5.1",
```
Comment (igordmn) [inline compose/compiler/compiler-hosted/src/main/java/androidx/compose/compiler/plugins/kotlin/VersionChecker.kt:124]: Could you push a git tag `jetpack-compose/1.5.1` that points to the Jetpack Compose Release?

image

File: compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/CoreTextField.kt (hunk 1 above)
Comment (igordmn) [reply compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/text/CoreTextField.kt:307]: This change disables input methods. It shouldn't break it for desktop/ios. I checked desktop with enabled state - it still works the same.

**Review by igordmn — APPROVED** [review]

**Comment by igordmn** [comment]
1. This PR should be merged, not squashed (to preserve history)
2. We also need to change versions [here](https://github.com/JetBrains/compose-multiplatform-core/blob/4ec63830db7f1db517f6e7119639e75b224248ef/gradle.properties#L103)

**Comment by eymar** [comment]
> This PR should be merged, not squashed (to preserve history)

~Enabled the "Create a merge commit" feature. It was disabled.~

UPD: reverted it back after Igor's comment

**Review by eymar — APPROVED** [review]

**Comment by igordmn** [comment]
> Enabled the "Create a merge commit" feature. It was disabled.

Better to disable it, as the state of the button is remembered, and we can forget to switch to Squash for usual PR's.

The merges should be done locally, pushing this branch directly to jb-main. The PR automatically closes after this.

**Pull request merged**